File: src/grpc_test.py
from __future__ import print_function
import grpc
import message_pb2
import message_pb2_grpc

def run():
  with grpc.insecure_channel('localhost:50051') as channel:
    stub = message_pb2_grpc.GreetServiceStub(channel)
    response = stub.SayHello(message_pb2.HelloRequest(name='World'))
    print("Greeter client received: " + response.message)

if __name__ == '__main__':
    run()

File: src/grpc_test.cpp
#include "message.grpc.pb.h"
#include <grpcpp/grpcpp.h>

using grpc::Server;
using grpc::ServerBuilder;
using grpc::ServerContext;
using grpc::Status;
using example::GreetService;
using example::HelloRequest;
using example::HelloResponse;

class GreetServiceImpl final : public GreetService::Service {
  Status SayHello(ServerContext* context, const HelloRequest* request, HelloResponse* response) override {
    std::string prefix("Hello ");
    response->set_message(prefix + request->name());
    return Status::OK;
  }
};

void RunServer() {
  std::string server_address("0.0.0.0:50051");
  GreetServiceImpl service;

  ServerBuilder builder;
  builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
  builder.RegisterService(&service);
  std::unique_ptr<Server> server(builder.BuildAndStart());
  std::cout << "Server listening on " << server_address << std::endl;
  server->Wait();
}
